### PR TITLE
PathPlanner-2.0: Merge Request

### PR DESCRIPTION
**Pull request by Omkaarcodes**
In the PathPlanner 2.0 Branch, I added a followPathCommand in SwerveSubsystem that uses PathPlannerLib to follow a path given a .path file (in deploy/pathplanner/paths). I also initialized a chooser object in RobotContainer (pathChooser) and made it such that you can choose which path to run. The selected command is returned in Robot.java under the autonomousinit() command.

Also, there's a JSON file for the PathPlannerLib as well.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java
Comment (lazychino) [inline src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java:80]: fix indentation

File: src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java
Comment (lazychino) [inline src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java:91]: I dont think `closestAngle` is being used.

File: src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java
Comment (lazychino) [inline src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java:122]: In here is more clear to just use POSE_ESTIMATOR::getEstimatedPose

File: src/main/java/frc/robot/RobotContainer.java
Comment (lazychino) [inline src/main/java/frc/robot/RobotContainer.java:96]: I don't think code should be calling periodic methods these should be run by the scheduler.

File: src/main/java/frc/robot/RobotContainer.java
Comment (lazychino) [inline src/main/java/frc/robot/RobotContainer.java:98]: why not get these from the Pigeon when needed?

File: src/main/java/frc/robot/RobotContainer.java
Comment (lazychino) [inline src/main/java/frc/robot/RobotContainer.java:48]: Why is this state needed here?

File: src/main/java/frc/robot/RobotContainer.java
Comment (lazychino) [inline src/main/java/frc/robot/RobotContainer.java:46]: These are not Constants

**Review by lazychino — CHANGES_REQUESTED** [review]
I think this might need to wait until #12 is merge and you can sync with main, so that is clear what are your changes vs the InputManager ones.

File: src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java
Comment (Omkaarcodes) [reply src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java:122]: The pathplannerlib documentation says to only reference the subsystem the method is on for any functions needed.

File: src/main/java/frc/robot/RobotContainer.java
Comment (Omkaarcodes) [reply src/main/java/frc/robot/RobotContainer.java:96]: So I don't need to call them for them to run? Will the PoseEstimator periodic function run automatically?

File: src/main/java/frc/robot/RobotContainer.java
Comment (Omkaarcodes) [reply src/main/java/frc/robot/RobotContainer.java:98]: That's there to update xVel, yVel, and rotationalVel later on if the sliders for them are used.

File: src/main/java/frc/robot/RobotContainer.java
Comment (Omkaarcodes) [reply src/main/java/frc/robot/RobotContainer.java]: I accidentally used constants, but I meant to say that they were important variables.

File: src/main/java/frc/robot/RobotContainer.java
Comment (lazychino) [reply src/main/java/frc/robot/RobotContainer.java:96]: That's my understanding. https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Subsystem.html#periodic()

File: src/main/java/frc/robot/RobotContainer.java
Comment (lazychino) [reply src/main/java/frc/robot/RobotContainer.java:98]: I dont see any sliders, are they defined in another place?

**Comment by lazychino** [comment]
@REVSIX please don't apply changes for the input manager branch here, that will create conflict when we merge.

**Review by lazychino — APPROVED** [review]
Handle issues on by video call

Approved pending that the project builds.

**Pull request merged**
